### PR TITLE
SM-768: Update EFCore and related packages to >= 7.0

### PR DESCRIPTION
**Pull request by coltonhurst**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

**Dependency upgrades for the following projects**:

- Infrastructure.EntityFramework ([link](https://github.com/bitwarden/server/pull/3006/files#diff-8e6e5d93455dc0d43dd5d41d96acd181ab5829bc8af609fa6a5fec0a42c2b581))
  - [Microsoft.EntityFrameworkCore.Relational](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Relational/7.0.5) -> 7.0.5
  - [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/7.0.5) -> 7.0.5
  - [Microsoft.EntityFrameworkCore.Sqlite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite/7.0.5) -> 7.0.5
  - [Npgsql.EntityFrameworkCore.PostgreSQL](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL/7.0.4) -> 7.0.4
  - [Pomelo.EntityFrameworkCore.MySql](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql/7.0.0) -> 7.0.0
  - [linq2db.EntityFrameworkCore](https://www.nuget.org/packages/linq2db.EntityFrameworkCore/7.5.0) -> 7.5.0

- Infrastructure.IntegrationTest ([link](https://github.com/bitwarden/server/pull/3006/files#diff-cfde4ea57f9f8ea8f8cae3fa3c0717e91f07bf3a79728f36925f66f0da12c146))
  - [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/7.0.0) -> 7.0.0
  - [Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection/7.0.0) -> 7.0.0
  - [Microsoft.Extensions.DependencyInjection.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.Abstractions/7.0.0) -> 7.0.0
  - [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0

- IntegrationTestCommon ([link](https://github.com/bitwarden/server/pull/3006/files#diff-aef8f07606cd2d809cc1a273545168b6b6f29c7f845ce23000062ddc9dfd3942))
  - [Microsoft.EntityFrameworkCore.InMemory](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.InMemory/7.0.5) -> 7.0.5

- ~~util/Migrator ([link](https://github.com/bitwarden/server/pull/3006/files#diff-3dd39a1df065ae7da8d90d6d543fc42223406bf5f7e58a510427b4fa50b67456))~~
  - ~~[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0~~

- ~~util/MsSqlMigratorUtility ([link](https://github.com/bitwarden/server/pull/3006/files#diff-bad4eb5f7934bf7613aa78b1341ebd2852e474c083b0ee99747bdb2b5b75ca9c))~~
  - ~~[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0~~
  - ~~[Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console/7.0.0) -> 7.0.0~~

- util/MySqlMigrations ([link](https://github.com/bitwarden/server/pull/3006/files#diff-274507c36877ab07418be6346e4ef0d0b4dd2fcc547531df6e198f45dda8e6cc))
  - [Microsoft.EntityFrameworkCore.Design](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Design/7.0.5) -> 7.0.5

- util/PostgresMigrations ([link](https://github.com/bitwarden/server/pull/3006/files#diff-e9197ece6de0fdc5bf34fdc4e293af16ebf09f374b4899944cfb3060fecd99e2))
  - [Microsoft.EntityFrameworkCore.Design](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Design/7.0.5) -> 7.0.5

- ~~util/Setup ([link](https://github.com/bitwarden/server/pull/3006/files#diff-1eaedf8566cd84fe82c52f793e266dc1360a8203880e3132265fddbcb19d3009))~~
  - ~~[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0~~

- util/SqlServerEFScaffold ([link](https://github.com/bitwarden/server/pull/3006/files#diff-16c7d87ed90e205b1f5932fcdda2df84c134dc0eade70a46e7b812dbc4eef332))
  - [Microsoft.EntityFrameworkCore.Design](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Design/7.0.5) -> 7.0.5

- util/SqliteMigrations ([link](https://github.com/bitwarden/server/pull/3006/files#diff-7fe931a4a7b5a7d427d178ff6ee199d4eeae0f6de8126e48baff9617b862fcde))
  - [Microsoft.EntityFrameworkCore.Design](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Design/7.0.5) -> 7.0.5

**Please note, the following tests are currently failing on master and are still failing after this change**:

- Infrastructure.EFIntegration.Test > Bit.Infrastructure.EFIntegration.Test
  - Auth.Repositories
    - EmergencyAccessRepositoryTests
      - CreateAsync_Works_DataMatches [failing]
  - Repositories
    - OrganizationRepositoryTests
      - CreateAsync_Works_DataMatches [failing]
      - GetByIdentifierAsync_Works_DataMatches [failing]
      - ReplaceAsync_Works_DataMatches [failing]
    - OrganizationSponsorshipRepositoryTests
      - CreateAsync_Works_DataMatches [failing]
      - ReplaceAsync_Works_DataMatches [failing]
    - UserRepositoryTests
      - ReplaceAsync_Works_DataMatches [failing]
      - UpdateRenewalReminderDateAsync_Works_DataMatches [failing]
  - Tools.Repositories
    - SendRepositoryTests
      - CreateAsync_Works_DataMatches [failing]

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: test/Infrastructure.IntegrationTest/Infrastructure.IntegrationTest.csproj [hunk 1]
```diff
@@ -7,10 +7,10 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.1" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="7.0.0" />
```
Comment (Hinton) [inline test/Infrastructure.IntegrationTest/Infrastructure.IntegrationTest.csproj:13]: Do we have to worry about potentially upgrading asp.net to 7, or are they backwards compatible?

**Review by Thomas-Avery — COMMENTED** [review]
util/Migrator ([link](https://github.com/bitwarden/server/pull/3006/files#diff-3dd39a1df065ae7da8d90d6d543fc42223406bf5f7e58a510427b4fa50b67456))

[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0

util/MsSqlMigratorUtility ([link](https://github.com/bitwarden/server/pull/3006/files#diff-bad4eb5f7934bf7613aa78b1341ebd2852e474c083b0ee99747bdb2b5b75ca9c))

[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0
[Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console/7.0.0) -> 7.0.0

util/Setup ([link](https://github.com/bitwarden/server/pull/3006/files#diff-1eaedf8566cd84fe82c52f793e266dc1360a8203880e3132265fddbcb19d3009))

[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0


I didn't think these projects were related to entity framework.

Was there something that came up for needing to update the dependency in these three projects?

**Comment by coltonhurst** [comment]
> util/Migrator ([link](https://github.com/bitwarden/server/pull/3006/files#diff-3dd39a1df065ae7da8d90d6d543fc42223406bf5f7e58a510427b4fa50b67456))
> 
> [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0
> 
> util/MsSqlMigratorUtility ([link](https://github.com/bitwarden/server/pull/3006/files#diff-bad4eb5f7934bf7613aa78b1341ebd2852e474c083b0ee99747bdb2b5b75ca9c))
> 
> [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0 [Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console/7.0.0) -> 7.0.0
> 
> util/Setup ([link](https://github.com/bitwarden/server/pull/3006/files#diff-1eaedf8566cd84fe82c52f793e266dc1360a8203880e3132265fddbcb19d3009))
> 
> [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/7.0.0) -> 7.0.0
> 
> I didn't think these projects were related to entity framework.
> 
> Was there something that came up for needing to update the dependency in these three projects?

No, everything builds correctly without these being updated. I updated them for consistency to stay in parallel with the `Infrastructure.IntegrationTest` project.

**Review by Thomas-Avery — DISMISSED** [review]
LGTM

**Review by jlf0dev — APPROVED** [review]
Looks good from an Auth perspective

**Review by Hinton — DISMISSED** [review]
Looks good from platform.

**Review by r-tome — DISMISSED** [review]
Looks good!

**Review by Thomas-Avery — DISMISSED** [review]

**Review by r-tome — DISMISSED** [review]

**Review by Hinton — DISMISSED** [review]

**Review by jlf0dev — APPROVED** [review]
👍

**Review by jlf0dev — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Review by Hinton — APPROVED** [review]

**Pull request merged**